patchwork-capabilities/src/main/java/com/patchworkmc/mixin/capability/ChunkSerializerMixin.java

```diff
@@ -0,0 +1,59 @@
+package com.patchworkmc.mixin.capability;
+
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.Slice;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.block.Block;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.structure.StructureManager;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.village.PointOfInterestStorage;
+import net.minecraft.world.ChunkSerializer;
+import net.minecraft.world.TickScheduler;
+import net.minecraft.world.World;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.chunk.Chunk;
+import net.minecraft.world.chunk.ChunkSection;
+import net.minecraft.world.chunk.UpgradeData;
+import net.minecraft.world.chunk.WorldChunk;
+
+import com.patchworkmc.impl.capability.CapabilityProviderHolder;
+
+@Mixin(ChunkSerializer.class)
+public class ChunkSerializerMixin {
```

Comment:
These mixins seems very specific but fragile. Is there not a better way?

Reply:
Up for grabs, if anyone wants to find a better way.